src/test/groovy/graphql/execution/instrumentation/TestBatchLoadingSupported.groovy

```diff
@@ -0,0 +1,206 @@
+package graphql.execution.instrumentation
+
+import graphql.ExecutionInput
+import graphql.ExecutionResult
+import graphql.GraphQL
+import graphql.StarWarsData
+import graphql.TypeResolutionEnvironment
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionStrategyParameters
+import graphql.schema.DataFetcher
+import graphql.schema.DataFetchingEnvironment
+import graphql.schema.GraphQLObjectType
+import graphql.schema.StaticDataFetcher
+import graphql.schema.TypeResolver
+import graphql.schema.idl.MapEnumValuesProvider
+import graphql.schema.idl.RuntimeWiring
+import org.dataloader.BatchLoader
+import org.dataloader.DataLoader
+import spock.lang.Specification
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.stream.Collectors
+
+import static graphql.TestUtil.schemaFile
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring
+
```

Comment:
This is interesting in that its quasi testing another library, that is the java-dataloader. 

 But it really demonstrates that the lower level `graphql-java` library is able to compose in behaviours via `Instrumentation`.

The alternative to this whole PR would be to have people have custom derived execution strategies where as this allows composition over inheritance.  Any `ExecutionStrategy` can be used and the behaviour can be inserted via Instrumentation.